File: app/src/main/java/me/jacobrr/whatsappanalyzer/FilteredFilePicker.java
package me.jacobrr.whatsappanalyzer;

import android.support.annotation.NonNull;
import com.nononsenseapps.filepicker.FilePickerFragment;

import java.io.File;

/**
 * Created by jacob on 12/11/2015.
 */
public class FilteredFilePicker extends FilePickerFragment {
    // File extension to filter on
    private static final String EXTENSION = ".txt";

    /**
     * @param file
     * @return The file extension. If file has no extension, it returns null.
     */
    private String getExtension(@NonNull File file) {
        String path = file.getPath();
        int i = path.lastIndexOf(".");
        if (i < 0) {
            return null;
        } else {
            return path.substring(i);
        }
    }

    @Override
    protected boolean isItemVisible(final File file) {
        if (!isDir(file) && (mode == MODE_FILE || mode == MODE_FILE_AND_DIR)) {
            return EXTENSION.equalsIgnoreCase(getExtension(file));
        }
        return isDir(file);
    }
}

File: app/src/main/java/me/jacobrr/whatsappanalyzer/FilteredFilePicker.kt
package me.jacobrr.whatsappanalyzer

import com.nononsenseapps.filepicker.AbstractFilePickerFragment
import com.nononsenseapps.filepicker.FilePickerFragment

import java.io.File

/**
 * Created by jacob on 12/11/2015.
 */
class FilteredFilePicker : FilePickerFragment() {

    /**
     * @param file
     * @return The file extension. If file has no extension, it returns null.
     */
    private fun getExtension(file: File): String? {
        val path = file.path
        val i = path.lastIndexOf(".")
        return if (i < 0) {
            null
        } else {
            path.substring(i)
        }
    }

    override fun isItemVisible(file: File): Boolean {
        return if (!isDir(file) && (mode == AbstractFilePickerFragment.MODE_FILE || mode == AbstractFilePickerFragment.MODE_FILE_AND_DIR)) {
            EXTENSION.equals(getExtension(file)!!, ignoreCase = true)
        } else isDir(file)
    }

    companion object {
        // File extension to filter on
        private const val EXTENSION = ".txt"
    }
}
